x/farming/types/keys.go

```diff
@@ -77,6 +87,23 @@ func GetQueuedStakingByFarmerPrefix(farmerAcc sdk.AccAddress) []byte {
 	return append(QueuedStakingIndexKeyPrefix, address.MustLengthPrefix(farmerAcc)...)
 }
 
+// GetQueuedStakingsByFarmerAndDenomPrefix returns a key prefix used to
+// iterate queued stakings by farmer address and staking coin denom.
+func GetQueuedStakingsByFarmerAndDenomPrefix(farmerAcc sdk.AccAddress, stakingCoinDenom string) []byte {
+	return append(
+		append(
+			QueuedStakingIndexKeyPrefix, address.MustLengthPrefix(farmerAcc)...),
+		LengthPrefixString(stakingCoinDenom)...)
+}
+
+// GetQueuedStakingEndBytes returns end bytes for iteration of queued stakings.
+// The returned end bytes should be used directly, not through
+// sdk.InclusiveEndBytes.
+// The range this end bytes form includes queued stakings with same endTime.
+func GetQueuedStakingEndBytes(endTime time.Time) []byte {
+	return append(QueuedStakingKeyPrefix, LengthPrefixTimeBytes(endTime.Add(1))...)
```

Comment:
What is the purpose of adding 1 to `endTime`?

Reply:
`GetQueuedStakingEndBytes` works like `sdk.InclusiveEndBytes`.
The range formed by this end bytes should include the `endTime` itself, and this is the purpose of `.Add(1)` here.
If we use `endTime` directly, then keys with a form of `{endTime}{stakingCoinDenom}{farmerAcc}` would be excluded from the range, since the key is behind sole `{endTime}` key.

Reply:
Maybe we could just use `sdk.PrefixEndBytes`.
Ref: https://github.com/cosmos/cosmos-sdk/blob/9f543b1c3b3d51e8f310fa9ca848d6b101c93f4c/x/gov/keeper/keeper.go#L189-L193

Reply:
Thanks for the explanation. I get it now. That is clear.